src/main/java/com/powsybl/openloadflow/dc/DcLoadFlowEngine.java

```diff
@@ -59,10 +55,18 @@ public DcLoadFlowEngine(List<LfNetwork> networks, DcLoadFlowParameters parameter
         this.parameters = Objects.requireNonNull(parameters);
     }
 
-    private void distributeSlack(LfNetwork network) {
-        double mismatch = network.getActivePowerMismatch();
+    protected void distributeSlack(Collection<LfBus> buses) {
+        double mismatch = getActivePowerMismatch(buses);
         ActivePowerDistribution activePowerDistribution = ActivePowerDistribution.create(parameters.getBalanceType(), false);
-        activePowerDistribution.run(network, mismatch);
+        activePowerDistribution.run(buses, mismatch);
+    }
+
+    public static double getActivePowerMismatch(Collection<LfBus> buses) {
```

Comment:
I don't like the idea of having as argument a list of buses because before it was a network that represents a unique synchronous component and now you hide that. Do you have to protect/private the functions that use this list as argument.